src/CollectionDataTable.php

```diff
@@ -24,6 +24,32 @@ class CollectionDataTable extends DataTableAbstract
     public $original;
 
     /**
+     * Can the DataTable engine be created with these parameters.
+     *
+     * @param mixed $source
+     * @return boolean
+     */
+    public static function canCreate($source)
+    {
+        return is_array($source) || $source instanceof Collection;
+    }
+
+    /**
+     * Factory method, create and return an instance for the DataTable engine.
+     *
+     * @param array|\Illuminate\Support\Collection $source
+     * @return \Illuminate\Support\Collection
```

Comment:
I think the correct return doc here is `CollectionDataTable|DataTableAbstract`?

Reply:
I believe not, the `|` [indicates](https://docs.phpdoc.org/references/phpdoc/tags/return.html) that either of those two types can returned but in this case it always returns a CollectionDataTable (which just happens to be an instance of DataTableAbstract). I actually think you shouldnt use DataTableAbstract as a type anywhere as an abstract class can never be instantiated and its always the parent class you should list. But correct me if I am wrong..

Reply:
Not really sure too. Sometimes, scrutinizer complains on things like this. Using PHPStorm, it will suggest to return both so I often go towards it. :)

Reply:
Thats probably because it doesnt understand that with the `parent::create` we call a method of the abstract class which uses the `static` keyword to return an instance of the current class.

The following code will probably trick PHPStorm to get it right:
```php
$datatable = parent::create(source);
/* @var $datatable CollectionDataTable */
return $datatable;
```
Should I use that?

Reply:
Maybe `/* @var $datatable self */` works too btw

Reply:
I think it is safe to use `CollectionDataTable|DataTableAbstract`. This way, IDE like phpstorm won't complain since it is compatible on the parent class as it returns `DataTableAbstract`.

Reply:
Yes boss, done and done :smile: